src/main/java/org/candlepin/subscriptions/tally/job/CaptureSnapshotsTaskManager.java

```diff
@@ -125,6 +130,31 @@ public void tallyAccountByHourly(String accountNumber, String startDateTime, Str
             .setSingleValuedArg("endDateTime", endDateTime).build());
     }
 
+    @Transactional
+    public void updateHourlySnapshotsForAllAccounts() {
+        try (Stream<String> accountStream = accountListSource.syncableAccounts()) {
+            AtomicInteger count = new AtomicInteger(0);
+
+            accountStream.forEach(accountNumber -> {
+
+                Duration metricRange = applicationProperties.getMetricLookupRangeDuration();
+
+                OffsetDateTime endTime = OffsetDateTime.now();
+                OffsetDateTime startTime = endTime.minus(metricRange);
```

Comment:
Two thoughts:
1. Let's define start/endtime outside of the for-loop
2. Let's normalize the time (I think truncating `now()` to the hour would probably do it)